### PR TITLE
feat added support for DOMAIN types

### PR DESCRIPTION
**Pull request by jimezesinachi**
# SynthQL pull request template

## Why

Adding support for domain types

## What changed

Logic in `introspect/generate` and generated types and schemas in the packages (auto-generated)

### CODE REVIEW

File: packages/introspect/src/generate.ts [hunk 1]
```diff
@@ -261,16 +262,27 @@ function createDomainJsonSchema(
...
+                `The ${domain.name} domain from the ${domain.schemaName} schema`,
         };
         return acc;
     }, empty);
 }
 
+function domainType(
+    domainInnerType: DomainDetails['innerType'],
+): JSONSchema['type'] {
+    if (domainInnerType === 'pg_catalog.int4') {
```
Comment (fhur) [inline packages/introspect/src/generate.ts:277]: This looks roughly right to me, I guess we should iterate over the "createWellKnownDefs" type to ensure we're supporting all native PG types.

**Review by fhur — CHANGES_REQUESTED** [review]
Good progress @jimezesinachi  👍 

Iterate over all known types and I think we will be done with this one.

File: packages/introspect/src/generate.ts (hunk 1 above)
Comment (jimezesinachi) [reply packages/introspect/src/generate.ts:277]: Makes sense. On it

**Review by fhur — APPROVED** [review]
LGTM 🚀

**Pull request merged**